lab5/utils/CollectionManager.java

```diff
@@ -0,0 +1,390 @@
+            for (City city : cities) {
+                String cityString = city.toString();
+                System.out.println(cityString);
+                System.out.println();
+            }
+        }
+    }
+
+    public void info() {
+        System.out.println("Information about collection");
+        System.out.println("Type: " + cities.getClass());
+        String initializationDate = getInitializationDate();
+        System.out.println("Initialization date: " + initializationDate);
+        int numberOfElements = cities.size();
+        System.out.println("Number of cities: " + numberOfElements);
+    }
+
+    private String getInitializationDate() {
+        String date = "";
+        try {
+            Path path = Paths.get("storage.xml");
+            BasicFileAttributes attrs = Files.readAttributes(path, BasicFileAttributes.class);
+            FileTime fileTime = attrs.creationTime();
+            Instant instant = fileTime.toInstant();
+            LocalDateTime localDateTime = LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm:ss");
+            date = localDateTime.format(formatter);
+
+        } catch (IOException e) {
+            System.out.println("exception: " + e.getMessage());
+        }
+        return date;
+    }
+
+    public void addCity() {
+        Scanner scanner = new Scanner(System.in);
+        City newCity = readCityFromScanner(scanner);
+        add(newCity);
+    }
+
+    public City readCityFromScanner(Scanner scanner) {
+        int id = generateId();
+        System.out.println("ID automatically generated: " + id);
+
+        System.out.print("Enter name: ");
+        String name = scanner.next();
+
+        Coordinates coordinates = addCoordinate(scanner);
+
+        LocalDate creationDate = LocalDate.now();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        String formattedDate = creationDate.format(formatter);
+        System.out.println("Creation date was automatically generated: " + formattedDate);
+
+        float area = addArea(scanner);
+        Long population = addPopulation(scanner);
+        double metersAboveSeaLevel = addMetersAboveSeaLevel(scanner);
+        float agglomeration = addAgglomeration(scanner);
+        Climate climate = addClimate(scanner);
+        Government government = addGovernment(scanner);
+        Human governor = addGovernor(scanner);
+
+        return new City(id, name, coordinates, creationDate, area, population, metersAboveSeaLevel, agglomeration, climate, government, governor);
+
+    }
+
+
+    private int generateId() {
+        Random random = new Random();
+        int id = random.nextInt(Integer.MAX_VALUE) + 1;
+        for (City city : cities) {
+            while (city.getId() == id) {
+                id = random.nextInt(Integer.MAX_VALUE) + 1;
+            }
+        }
+        return id;
+    }
+
```

Comment:
Lets say you have two cities, with ids 1 and 2

The first random gives you 2, then you loop until the second city, in the second random it gives you 1, and you return it

So you wil have two cities with the id 1

This algorithm is not only incorrect and terrible waste of cpu time, why not just auto-increment?